src/renderers/dom/shared/DOMPropertyOperations.js

```diff
@@ -158,6 +158,12 @@ var DOMPropertyOperations = {
         } else if (propertyInfo.hasBooleanValue ||
                    (propertyInfo.hasOverloadedBooleanValue && value === true)) {
           node.setAttribute(attributeName, '');
+        } else if (attributeName === 'value' && node.hasAttribute('value')) {
+          // Use loose coercion to prevent replacement on comparisons like
+          // '3e1' == 30 in Chrome (~52).
+          if (node.value != value) { // eslint-disable-line
+            node.setAttribute(attributeName, '' + value);
```

Comment:
React does deviate from browser behavior when it comes to controlled components (e.g., change events on keydown instead of on blur). Currently the `value` attribute does update on controlled inputs, and it looks like we're explicitly setting it, so there may be a good reason I'm not aware of.

cc @spicyj @zpao